lib/elixir/lib/registry.ex

```diff
@@ -1435,17 +1442,18 @@ defmodule Registry.Partition do
 
   # The key partition is a set for unique keys,
   # duplicate bag for duplicate ones.
-  defp init_key_ets(:unique, key_partition) do
-    :ets.new(key_partition, [:set, :public, read_concurrency: true, write_concurrency: true])
+  defp init_key_ets(:unique, key_partition, compressed) do
+    opts = [:set, :public, read_concurrency: true, write_concurrency: true]
+    :ets.new(key_partition, compression_opt(opts, compressed))
   end
 
-  defp init_key_ets(:duplicate, key_partition) do
-    :ets.new(key_partition, [
-      :duplicate_bag,
-      :public,
-      read_concurrency: true,
-      write_concurrency: true
-    ])
+  defp init_key_ets(:duplicate, key_partition, compressed) do
+    opts = [:duplicate_bag, :public, read_concurrency: true, write_concurrency: true]
+    :ets.new(key_partition, compression_opt(opts, compressed))
+  end
+
+  defp compression_opt(opts, compressed) do
+    if compressed, do: [:compressed] ++ opts, else: opts
```

Comment:
nit:  `[:compressed | opts]` should be slightly more performant and idiomatic?

Reply:
Both are equivalent when the left side of `++` is a literal. :)